node-js/src/middleware/jwt-middleware.js

```diff
@@ -0,0 +1,123 @@
+        4- Check the token_use claim.
+            1- If you are only accepting the access token in your web API operations, its value must be access.
+            2- If you are only using the ID token, its value must be id.
+            3- If you are using both ID and access tokens, the token_use claim must be either id or access.
+        # You can now trust the claims inside the token.
+        */
+        // decode token
+        var decodedAccessJwt = jwt.decode(access_token, { complete: true });
+        if (!decodedAccessJwt) {
+            throw { message: "Not a valid Access JWT token", status: 401 };
+        }
+        // #1 check token expiration
+        var currentTs = Math.floor(new Date() / 1000);
+        if (currentTs > decodedAccessJwt.payload.exp) {
+            throw { message: "Token expired", status: 401 };
+        }
+        // #2 check audience
+        var decodedIdJwt = jwt.decode(id_token, { complete: true });
+        if (!decodedIdJwt) {
+            throw { message: "Not a valid Id JWT token", status: 401 };
+        }
+        if (decodedIdJwt.payload.aud !== process.env.CLIENT_ID || decodedAccessJwt.payload.client_id !== process.env.CLIENT_ID) {
+            throw { message: "Token was not issued for this audience", status: 401 };
+        }
+        // #3 check issuer
+        const cognitoIssuer = `https://cognito-idp.${process.env.AWS_REGION}.amazonaws.com/${process.env.USER_POOL_ID}`;
+        if (decodedIdJwt.payload.iss !== cognitoIssuer || decodedAccessJwt.payload.iss !== cognitoIssuer) {
+            throw { message: "Token was not issued by this issuer", status: 401 };
+        }
+        // #4 check token_use
+        if (decodedAccessJwt.payload.token_use !== "access") {
+            throw { message: "Token was not an access token", status: 401 };
+        }
+        if (decodedIdJwt.payload.token_use !== "id") {
+            throw { message: "Token was not an id token", status: 401 };
+        }
+        // download tokens
+        const pems = await axios.get(`${cognitoIssuer}/.well-known/jwks.json`)
+            .then(function (response) {
+                var pems = {};
+                var keys = response.data['keys'];
+                for (var i = 0; i < keys.length; i++) {
+                    // Convert each key to PEM
+                    var key_id = keys[i].kid;
+                    var modulus = keys[i].n;
+                    var exponent = keys[i].e;
+                    var key_type = keys[i].kty;
+                    var jwk = { kty: key_type, n: modulus, e: exponent };
+                    var pem = jwkToPem(jwk);
+                    pems[key_id] = pem;
+                }
+                return pems;
+            })
+            .catch(function (error) {
+                // handle error
+                return undefined;
+            });
+        if (!pems) {
+            throw { message: "Internal Server Error: PEMS", status: 500 };
+        }
```

Comment:
```suggestion
            .catch(function (error) {
                throw { message: "Internal Server Error: PEMS", status: 500 };
            });
```

Esse `throw` poderia estar dentro do `catch` certo?